src/rust/engine/process_execution/src/remote.rs

```diff
@@ -384,6 +400,53 @@ mod tests {
     );
   }
 
+  #[test]
+  fn wait_between_request() {
+    // wait at least 500 milli for one retry
+    {
```

Comment:
Rather than making separate scopes within the one test, let's just make these two separate tests